Password recovery broken on codenvy.io

As per subject. Post requests returns 500

Trace
```
2017-02-03 13:19:18,785[nio-8080-exec-1]  [ERROR] [c.c.s.password.PasswordService 161]  - Error during setting user's password

java.io.IOException: Failed access: http://localhost:8080/mail/mail/send?token, method: POST, response code: 500, message: Can't read from input stream org.everrest.core.impl.provider.json.JsonException: Unexpected end of stream.

        at org.eclipse.che.api.core.rest.DefaultHttpJsonRequest.doRequest(DefaultHttpJsonRequest.java:280) ~[che-core-api-core-5.2.1.jar:5.2.1]

        at org.eclipse.che.api.core.rest.DefaultHttpJsonRequest.request(DefaultHttpJsonRequest.java:159) ~[che-core-api-core-5.2.1.jar:5.2.1]

        at com.codenvy.mail.MailSenderClient.sendMail(MailSenderClient.java:74) ~[codenvy-hosted-mail-client-5.2.1.jar:5.2.1]

        at com.codenvy.service.password.PasswordService.recoverPassword(PasswordService.java:156) ~[codenvy-hosted-recoverpassword-5.2.1.jar:5.2.1]

        at sun.reflect.GeneratedMethodAccessor388.invoke(Unknown Source) ~[na:na]

        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_102]

        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_102]

        at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:140) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:60) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.RequestDispatcher.doInvokeResource(RequestDispatcher.java:306) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.RequestDispatcher.invokeSubResourceMethod(RequestDispatcher.java:297) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:233) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:128) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.RequestHandlerImpl.handleRequest(RequestHandlerImpl.java:62) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.impl.EverrestProcessor.process(EverrestProcessor.java:120) [everrest-core-1.13.1.jar:na]

        at org.everrest.core.servlet.EverrestServlet.service(EverrestServlet.java:61) [everrest-core-1.13.1.jar:na]

        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729) [servlet-api.jar:na]

        at com.google.inject.servlet.ServletDefinition.doServiceImpl(ServletDefinition.java:286) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:276) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:181) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:85) [guice-servlet-4.1.0.jar:na]

        at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:436) [catalina.jar:8.0.38]

        at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:177) [catalina.jar:8.0.38]

        at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120) [guice-servlet-4.1.0.jar:na]

        at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:135) [guice-servlet-4.1.0.jar:na]

        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) [catalina.jar:8.0.38]

        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) [catalina.jar:8.0.38]

        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212) [catalina.jar:8.0.38]

        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) [catalina.jar:8.0.38]

        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141) [catalina.jar:8.0.38]

        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [catalina.jar:8.0.38]

        at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:189) [logback-access-1.1.2.jar:na]

        at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:676) [catalina.jar:8.0.38]

        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) [catalina.jar:8.0.38]

        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:528) [catalina.jar:8.0.38]

        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1100) [tomcat-coyote.jar:8.0.38]

        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:687) [tomcat-coyote.jar:8.0.38]

        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520) [tomcat-coyote.jar:8.0.38]

        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476) [tomcat-coyote.jar:8.0.38]

        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_102]

        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_102]

        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.0.38]

        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_102]
```

--- a/wsmaster/codenvy-hosted-mail-server/pom.xml
+++ b/wsmaster/codenvy-hosted-mail-server/pom.xml
@@ -60,6 +60,10 @@
         </dependency>
         <dependency>
             <groupId>org.eclipse.che.core</groupId>
+            <artifactId>che-core-api-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.che.core</groupId>
             <artifactId>che-core-commons-inject</artifactId>
         </dependency>
         <dependency>

--- a/wsmaster/codenvy-hosted-mail-server/src/main/java/com/codenvy/mail/deploy/CloudIdeMailModule.java
+++ b/wsmaster/codenvy-hosted-mail-server/src/main/java/com/codenvy/mail/deploy/CloudIdeMailModule.java
@@ -17,10 +17,10 @@ package com.codenvy.mail.deploy;
 import com.codenvy.mail.MailSender;
 import com.google.inject.AbstractModule;
 
+import org.eclipse.che.api.core.rest.CoreRestModule;
 import org.eclipse.che.inject.DynaModule;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
 
 /**
  * Guice container configuration file. Replaces old REST application composers and servlet context listeners.
@@ -33,5 +33,6 @@ public class CloudIdeMailModule extends AbstractModule {
     @Override
     protected void configure() {
         bind(MailSender.class);
+        install(new CoreRestModule());
     }
 }

